internal/redfishwrapper/virtual_media.go

```diff
@@ -29,50 +33,43 @@ func (c *Client) SetVirtualMedia(ctx context.Context, kind string, mediaURL stri
 		return false, errors.New("invalid media type")
 	}
 
-	for _, manager := range managers {
-		virtualMedia, err := manager.VirtualMedia()
+	for _, m := range managers {
+		virtualMedia, err := m.VirtualMedia()
 		if err != nil {
 			return false, err
 		}
-		for _, media := range virtualMedia {
-			if media.Inserted {
-				err = media.EjectMedia()
-				if err != nil {
-					return false, err
-				}
-			}
+		if len(virtualMedia) == 0 {
+			return false, errors.New("no virtual media found")
 		}
-	}
 
-	// An empty mediaURL means eject everything, so if that's the case we're done. Otherwise, we
-	// need to insert the media.
-	if mediaURL != "" {
-		setMedia := false
-		for _, manager := range managers {
-			virtualMedia, err := manager.VirtualMedia()
-			if err != nil {
-				return false, err
+		for _, vm := range virtualMedia {
+			if vm.Inserted {
+				if err := vm.EjectMedia(); err != nil {
+					return false, err
+				}
 			}
-
-			for _, media := range virtualMedia {
-				for _, t := range media.MediaTypes {
-					if t == mediaKind {
-						err = media.InsertMedia(mediaURL, true, true)
-						if err != nil {
+			if mediaURL != "" {
+				if slices.Contains(vm.MediaTypes, mediaKind) {
+					if err := vm.InsertMedia(mediaURL, true, true); err != nil {
+						// Some BMC's (Supermicro SYS-E300-D9, for example) don't support the "inserted" and "writeProtected" properties,
```

Comment:
Can this comment be updated to to mainboard model number instead - X11SDV-4C-TLN2F,
The product name - SYS-* tends to change based on where the hardware is purchased and bmclib internally uses the mainboard model.

Reply:
ah yes, sorry about that. I will update it. thanks for looking up the mainboard model number!